geospaas/base_viewer/views.py

```diff
@@ -1,73 +1,51 @@
 from django.shortcuts import render
 from django.views.generic import View
 
-from geospaas.base_viewer.forms import TimeAndSourceForm, SpatialSearchForm
+from geospaas.base_viewer.forms import OverallForm
 from geospaas.catalog.models import Dataset as CatalogDataset
 
 # Create your views here.
 
 
 class IndexView(View):
 
-    form_class = [TimeAndSourceForm, SpatialSearchForm,]
+    form_class = OverallForm
     main_template = 'base_viewer/template_for_base.html'
     viewname = 'index'
-    #forms = list()
-
-    @classmethod
-    def create_forms(cls, *args):  # request_method='GET'
-        ''' Set default values for the form by instantiating them '''
-        return [i(*args) for i in cls.form_class]
 
     @classmethod
     def get_all_datasets(cls):
         return CatalogDataset.objects.all()
 
-    @staticmethod
-    def validate_forms(forms):
-        for form in forms:  # for loop for making clean data by is_valid() method
-            form.is_valid()
-        return forms
-
     @classmethod
-    def get_filtered_datasets(cls, forms):
+    def get_filtered_datasets(cls, form):
         ds = cls.get_all_datasets()
-        for form in forms:
-            # using the filter function of each form sequentially
-            ds = form.filter(ds)
-        return ds
+        return form.filter(ds)
 
     @classmethod
-    def set_context(cls, forms, ds):
+    def set_context(cls, form, ds):
         context = {}
-        # setting the forms based on their class into the correct naming in the context
-        for j in cls.form_class:
-            for form in forms:
-                if (isinstance(form.__class__(), j )):
-                    context[j.__name__] = form
-
+        context[cls.form_class.__name__] = form
```

Comment:
It is better not to hardcode here a name of the form class. Otherwise we have to hardcode the name of the form class in template and we will not be able to reuse the template.
```suggestion
        context['form'] = form
```